include/alpaka/vec/Vec.hpp

```diff
@@ -285,8 +287,21 @@ namespace alpaka
         ALPAKA_FN_HOST_ACC friend constexpr auto operator-(Vec const& p, Vec const& q) -> Vec
         {
             Vec r;
-            for(typename TDim::value_type i = 0; i < TDim::value; ++i)
-                r[i] = p[i] - q[i];
+#if BOOST_COMP_NVCC && BOOST_COMP_NVCC < BOOST_VERSION_NUMBER(11, 3, 0)
+            if(TDim::value > 0)
```

Comment:
Are older nvcc versions unable to consume `if constexpr`?

Reply:
Not in this specific instance, where you will get a syntax error with nvcc 11.2 and below. Generally, `if constexpr` is supported.